### ISSUE TITLE
String#replace with regex provides previous capture group when group not found

### ISSUE THREAD

**Issue by lauriro** [jerryscript-project/jerryscript#3743]
```javascript
var re = /'|{(.+)}/g
"{a}b'c".replace(re, function(match, group) {
	print(match, group)
})
```

jerryscript prints:

> {a} a
> ' a   <- notice `a` as captured group

expected:

> {a} a
> ' undefined

setup

> "Arch Linux" (Linux 5.5.9-arch1-2 x86_64)
> commit: 95aa82763501d0bf1592d121fc0e8af5066ee726
> build command: python tools/build.py

**Comment by dbatyai** [comment]
Will be resolved by #3746.

**Comment by lauriro** [comment]
I tested and this patch does resolve my issue, thank you! 👍

**Comment by rerobika** [comment]
Resolved via #3746 .
